### ISSUE TITLE
Qwant gibberish results

### ISSUE THREAD

**Issue by plant7403** [searxng/searxng#2563]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
I'm using nixos, and not sure how to get the version
At the bottom of the page it says 1.0.0

**How did you install SearXNG?**
Nixos, but the problem persisted in docker too
**What happened?**
Qwant search results are complete gibberish that lead to nonexistent websites

**How To Reproduce**
Search for anything

**Expected behavior**
Qwant gives English results

**Screenshots & Logs**
![Screenshot_20230720-113437_1](https://github.com/searxng/searxng/assets/24483751/aa668f41-ab1f-4f8c-8871-994ee1403c63)
And it's not just one result, it's many of them
Even more if you search for something less searchable

**Comment by return42** [comment]
> At the bottom of the page it says 1.0.0

Sure you installed SearXNG and not searx?

**Comment by plant7403** [comment]
@return42 it says "Powered by searxng - 1.0.0 - ..."
And the main logo is SearXNG
P.S. I'm pretty sure the version is 1.0.0 because it poorly packaged (or how nixos itself works)
You can check everything yourself https://searx.egor.wtf/

**Comment by return42** [comment]
> You can check everything yourself https://searx.egor.wtf/

:+1:

> P.S. I'm pretty sure the version is 1.0.0 because it poorly packaged (or how nixos itself works)

Sorry, I do not have experience with nixos .. package search shows me:

![image](https://github.com/searxng/searxng/assets/554536/388a5612-3666-4b96-b04c-d767113c57cf)

In the nixos package repo I [see](https://github.com/NixOS/nixpkgs/blob/7363eedd89566bf14ff118dcb0cc187b30804bdc/pkgs/servers/web-apps/searxng/default.nix#L10-L15) commit da7c30291dcf53c is used for the package build .. not sure how instance updates do work in nixos but [SearXNG is a rolling release](https://docs.searxng.org/admin/update-searxng.html#migrate-and-stay-tuned) .. the deployment tool is git / git needs to be installed on the target host (I don't see a git requirement in the SearXNG package definitions of nixos).

To summarize: with nixos I can't help, but may be @SuperSandro2000, @999eagle or @nagy can help.

----

Additional I'm confused by this ...

>> How did you install SearXNG?

> Nixos, but the problem persisted in docker too

In a SearXNG docker installation you see version 1.0.0?

May be should focus on one installation method / we from SearXNG team prefer to use one of the [official installation methods](https://docs.searxng.org/admin/installation.html).

**Comment by plant7403** [comment]
> Additional I'm confused by this ...
> 
> How did you install SearXNG?
> 
> Nixos, but the problem persisted in docker too
> 
> In a SearXNG docker installation you see version 1.0.0?

Before jumping on nixos I used searxng in docker, and had it running in docker
And for the last year I had the same issue, I was just turning off qwant, but now I decided to find why it's the way it is

HOLD ON
I think I found the problem
Qwant isn't actually available in my country
I'm pretty sure other engines that were not supported (and now supported) before were giving an error or a timeout
But qwant is giving gibberish, could this be the cause?

**Comment by 999eagle** [comment]
@return42 Packages on NixOS should be reproducible, so we build SearXNG from a specific commit (da7c30291dcf53cc5b3d98f9aada5615cd1593a9 currently for the unstable branch as you've already seen yourself). Automatic upgrades to newer commits can't happen as packages, once built, are immutable.
With a cursory look through the diff between the currently packaged commit and master (https://github.com/searxng/searxng/compare/da7c30291dcf53cc5b3d98f9aada5615cd1593a9..master) I don't see any changes to the Qwant engine.

**Comment by return42** [comment]
@999eagle thanks for clarify :+1:

> But qwant is giving gibberish, could this be the cause?

with the hint from @999eagle and the hint that the results in docker are not better either, I suspect that the problem is outside the SearXNG installation.

> Qwant isn't actually available in my country

SearXNG uses the qwant's api: 

https://github.com/searxng/searxng/blob/a446dea1bb492eac417de9a900fae7cdf94aeec0/searx/engines/qwant.py#L70

https://github.com/searxng/searxng/blob/a446dea1bb492eac417de9a900fae7cdf94aeec0/searx/engines/qwant.py#L99-L100

When you send a API request from the IP of the SearXNG instance, do you get reasonable results for:

   https://api.qwant.com/v3/search/web?q=hello&locale=en_us

**Comment by SuperSandro2000** [comment]
We are just missing a step and the fallback version here is used https://github.com/searxng/searxng/blob/master/searx/version.py#L11-L16

The file is generated from https://github.com/searxng/searxng/blob/a446dea1bb492eac417de9a900fae7cdf94aeec0/searx/version.py#L119

@999eagle could you do a PR to include version and some dummy information for the git things? That would be awesome.

**Comment by return42** [comment]
Sadly the workflow to get a *frozen version* is undocumented / may @dalf can help us on how to generate a frozen version number (when the nixox package is build).

**Comment by return42** [comment]
@SuperSandro2000 @999eagle : from #229 ...

It is possible to call `python -m searx.version freeze` to freeze the current version (it creates the `searx/version_frozen.py` file)

It is useful when the code is install outside git (distro package, docker, etc...)

**Comment by plant7403** [comment]
> When you send a API request from the IP of the SearXNG instance, do you get reasonable results for:

```
{"status":"success","data":{"query":{"locale":"en_us","query":"hello","offset":0,"queryContext":{"originalQuery":"hello"}},"result":{"denied":0,"total":10,"items":{"mainline":[{"type":"web","count":10,"items":[{"title":"Waloj unatarva koztuli hello Ta li tarbu.","favicon":"","url":"http://bipuwanal.vg/buk","source":"http://edkeli.mx/dagnicla","desc":"Fa kemalfod faih tagocre lelav feakro jovuwdo keuk tih rutra komife um miza idlapu riwhil ta. Bof pibfukber fu ribe evigu roto ojvilvof zu zuv evpel nujwejdab wuvohov omrez pumoj govoido tezih. Apuulu aza oka ugulo da falgomah ku gadorepi rimnemiri nuvlo fa robwa jov satsik cifod inka. Mag kukusoba catibe cale jagwevse adocoobi wagzat luznilro rato fa tezefefug rios liv wolot wuf ri aju. Omnuf vu ragu bokgu ir himfa vosnip ipbodkes katbi efudoun utafiro izkauf rozgemi odu go.","_id":"6b8de0ca084550751ccb7d67c48fc10f"},{"title":"Jo newcuhot zimso pikace hello Kuwmab nim.","favicon":"","url":"http://ubebso.cv/gedo","source":"http://masozbaf.gp/ohoce","desc":"Sis waffufto depga biv zukserimo wuwlogda fa ce ja macsugnu fiogu hi oge. Nucteufo enefu damobno niwzon bahutve ofoduj ozi vij uketev uzo locugo manji jica lumikve. Anuhu docuvgoh tabo ho cihaz idizes pus acga efuto hiiwkek jipiwgof pelivuv lizonwa. Palejmo keri le costoso ka bi simu jac tec fosvupve webbojzep levwe safom am om bidfi ku cek. Cu mej sojkevij nori imjat jif roif fa juj nu idijibiz ninva feuc didew suvugnar. Moltiboz geoveiz latta donvescav antajfuc gisligum doejota zordowa hevkat heegduf zi ed cocanoc. Coje mu mup abumajta ob hahic li zaveworif ke ju ebcev iprihsa fitdozha voonu juwajosi.","_id":"dda125faa93c027aad8a64495d570933"},{"title":"Kohincut hello Le lasanga jed itenag no.","favicon":"","url":"http://cemmet.com/vagajilut","source":"http://re.kh/zafvi","desc":"Duhfogu osi of otaofa jusdezo nab eti dud umfofkub kazo pogilo pocul mocte bawaze. Jirli rifuba vuabhi katdag sera zumoziw suhanfuv ba sarli jif zoesauf ven. Huculpiz didavre towrefmo umpa rerzetde ce va lekemluc em uwumopu oclubsob le poaveogi ceizozo osusigiku ga luwlo. Voore lavli bafabco bujathim luerodo obi egoka uhe alenacide nil mi abitdit jibho guni beweb agaezoho. Ulifagke wo givuk liujbo ipupvu bubri uzefinjah ulase kuvlipzu epsuv kikoja od herek lalcice. Evsumi veza jo duzur ugamel mohecaji ejdu nuv hizmuho geh lajnof san.","_id":"ff48ec90c87402de8eacbd05124ad654"},{"title":"Dihlero obosuble hello Jalo lu tof puugi.","favicon":"","url":"http://muci.wf/ruwrunto","source":"http://kiputjis.ng/gutiefa","desc":"Lorza mag ag in je togap obu na lukso solgis imiakapi otejetaf serwojmam luftaper jufom duzecu zugasiv he. Ja zoc kijonafu ohoci nabeset apimole somuog ta vam recdic ajarid hare nigluwi dofmap. Ronob im raricne ni visfi porerkog ri ucihud ot lebvo zek lakweb goz je funow hogsu nol. Diforu uciaci fotkis hefore nuhewma de gu kezur vujhober cah bof galposuk. Pugimcun ce noege oheabo ripvieru zaskeuf tovkadpan wipco ez ecucov vefu jaw puhefpid ki.","_id":"fbb697b4124cc5c3f551fa4965955752"},{"title":"Da pikaz hello Wikro iro lukaiz biamced.","favicon":"","url":"http://sadhu.sn/volal","source":"http://be.vi/joki","desc":"Bo pigepfi amupalhu iwi coamdiw caturtit fah ikafekoh efotug ziezu faj zecic cezmapij guufe zew. Nuw volzuhfi masag bafhuzjo enainato pudom go lom kom bofeti awse divi jajsid lo. Cuwabka te vol jebago hatmu howe cepaja ril desuggo rimidho ugufo inwuv ecejuho juhdavdo af. Itous baw sib ew bukomu guiw dap bi dunkif goriv lijzipif to luh dal tiuvuev. Wiili hosu jovev mod madi lavmuj gaduzwi kaltuz wuru kipupdo mahe ecapizet gefjuhfis kevdujko wi voc puhanesa. Umeve ulke ise na topeb was orfobep tefwal bajso iku notvesan peto hew ro cuvaf mijafeguh. Famlam vaheka aco ra mima uli miwi uwa jov jas vejra lut cogapsoj wela facfas fe hiw eb.","_id":"b0892907d9cb74413d593c957627d622"},{"title":"Gakowna kevok hello Uwiro fal gamof fab.","favicon":"","url":"http://sih.ge/usize","source":"http://oc.bh/halogavu","desc":"Reb fovugwe uw tisiga ge tofviwnoc curopru beono dacu daombup liol munejcul wujeh fin un serebe ah. Uvuwa urajeder baseho anhi oki etelo emu ew bimiczer jug kid cajathu mesefuw akrec teuno buzrupu su. Im omejijib buhube ijiwsa nos bofa bej tenno tascovi ze ivi lokfijmi jok lop rurbinnam ila. Sussenas hufofze lelidso monobdo zesazje rakirre es buvpizu zal ziveb wer teir hed. Esged un sunlikgu zauj hat funisuf cefon bigi peho vuzfenho emivot mueba kos sewukmaz lodgez. Moki gocita amko ihoseji keb nuv duk wewunwop fa zuk hatuweaci liuginaz epuiju zupuj zaoh. Menwe belovilo issoilu vepaw bonurapo bigu cunu acfe di wipse gopu ezu oriset hah.","_id":"828091a3da1e5027af8e3c3527951b3f"},{"title":"Wiibo lit zij defu hello Boizi gal.","favicon":"","url":"http://hajiki.in/kebavit","source":"http://vebir.gd/ufajuzam","desc":"Uzamic zecices kejfon juw buw utat pom wawej wa ra velpuljup wozobneh. Mutub zap jivusiuh ebuho ce butesosew kijusahu pepim culsu kehakek rudcec pi wujitnu erawav et ti utierno. Doezihut hu visowzi zuwded fow keilmok norjej nojegtih li osinowgif pe gahinug osiov ta ep. Dilmowena badte zuzhiwuz amne vew goefovun pemmipa ru ule cuw emowah biiv zez widuc fazfopva doder tac zu. Fuj wa hag zub punogid cuvulahe muoso mezos saha fep mitsap rum oheluc ganepes ges ujhicef gipok. Suslo uhtiwu onu pe wa du wo juho fod cisih ozacetmud sapa uttige he ulu.","_id":"27af48c6ea9fcef6b37b0c9fb9af7469"},{"title":"Aha ejeozjo gicophi rugelle jinolra hello Mi.","favicon":"","url":"http://vuhmagegu.na/ju","source":"http://susiv.pr/iphopo","desc":"Lo ewkovos hocozgi jupes lizbo zenid oda baho putug lif okvepef kad bew wais nuceme vielo kohakpu. Kilzuw nih tud eca fikpef padlu norfiz nonatoj ganibagi amugenof emofadmej dam odjalod pif gokozi hohocva ovepuec. Nuz mafis ja pevagfog afa puol eti dodag rec hitara wumlamcef duhe aflalcuc atoohsaz rulebojo bobfowa lupiev. Gaden wedef irzajhe ucu pucalar idievme foguc tisfo av lene vimocohat wupuha. Ubfa katgace obuturop witduwbe tu icaizi hubawbu beswu ugoijaih cievazen honzaspa rodices jakigti. Kabecuwo pusfes hezav datot irwudpe hus vu habpac sahbe gi enuaduana nu herso hetefe. Rok bem ul orizo ristege magpih wi ciep kapukiloh fukasu zus uwfo wenahe taffarep hueje.","_id":"d7b35039b214d3062fd1e0bf35e6d7ea"},{"title":"Cicanac hello Zu riik wi cor gag.","favicon":"","url":"http://okruv.pk/he","source":"http://rizovze.cr/gudweedi","desc":"Hohpenmu em mah fag na coc podbuzan jodfurew tuowujur hi girokujop eruru rinig iwoc. Osli zev reg ebawoku uze co zisjawam ojje dorhujun zanca jej najetob ho niimo. Modi gajuwpud pebufzop ipfori di burvev oljohju sun je ajjo pufje gabre uc.","_id":"b4a3c2492fb72961f2275b04fe296a83"},{"title":"Gabu em hello Ralvumzu busesit leri lempiow.","favicon":"","url":"http://cuvwih.pg/netobwog","source":"http://bajki.ms/jeolakug","desc":"Judul hutfun to jawful eca ru bacu virisida jowisve fapfi viter kanulho vufu udda la ug fanit ero. Alikowo lopedruf lirgijoha iva mijiknod zap pujpud efraas tatoka cab guovgil hicek cohep fudduris vebmimpik. Atelril cu pukowre oh oc abijeflof gakuj pismivo mo sekaro tibirelef sar vub nego lempav rer leko. Zude at os hi com hor beb vizubajo gim ejuivib nob ibu tupmavof cuj hakocko iwu. Zarogwu geg ig tatudlu mikhur zaeloju kigahdet rufaf uzu jokwugce ujocudsug atahoiw wisaane ricnabven du. Ofekul narco pizta goju dur ritsazpi vawa saw rubduva vi oji usuf lubonhu ji or.","_id":"67d34339a940ea92d667dd9f0985a91a"}]}],"sidebar":[],"headline":[]},"ads":[],"lastPage":false,"filters":{"freshness":{"label":"Freshness","name":"freshness","type":"dropdown","selected":"all","values":[{"value":"all","label":"All time","translate":true},{"value":"day","label":"Past 24 hours","translate":true},{"value":"week","label":"Past week","translate":true},{"value":"month","label":"Past month","translate":true}]}}}}}
```

**Comment by return42** [comment]
@xxxcrow Yeah, seems gibberish what you got from qwant .. 

> Qwant isn't actually available in my country

just wondering that qwant responses gibberish on requests from Vietnam (14.177.55.154 / Ha Noi)

The only solution I see is to proxy the outgoing request or get an IP that is not located in a region qwant (for whatever reasons) dislike.

Related:

- [However, it is still possible for them to access Qwant in other parts of the world via VPN tools. Qwant is still available in more than 10 languages and in more than 30 countries. See you soon on Qwant. ](https://www.reddit.com/r/Qwant_se/comments/ke6pmt/new_info_regarding_the_unavailability_of_qwant_in/)
- https://www.reddit.com/r/Qwant_se/comments/tdbbus/why_qwant_isnt_working_in_some_countries/
- https://twitter.com/QwantCom/status/1339149434572206080

**Comment by 999eagle** [comment]
I've opened NixOS/nixpkgs#244473 to include the frozen version and update to current master.

**Comment by return42** [comment]
> I've opened [NixOS/nixpkgs#244473](https://github.com/NixOS/nixpkgs/pull/244473) to include the frozen version and update to current master.

@999eagle: Thanks :+1: I placed a comment https://github.com/NixOS/nixpkgs/pull/244473 .. the nixos discussion should be continued in your PR.

@xxxcrow: the issue you have is related to the qwant service itself .. see my comment from above https://github.com/searxng/searxng/issues/2563#issuecomment-1643693127 --> I'm going to close this github issue.
